site/src/pages/CreateTemplatePage/CreateTemplateForm.tsx

```diff
@@ -81,23 +84,31 @@ const defaultInitialValues: CreateTemplateData = {
   allow_user_autostart: false,
   allow_user_autostop: false,
   allow_everyone_group_access: true,
+  provisioner_type: "terraform",
 };
 
 type GetInitialValuesParams = {
   fromExample?: TemplateExample;
   fromCopy?: Template;
   variables?: TemplateVersionVariable[];
   allowAdvancedScheduling: boolean;
+  searchParams: URLSearchParams;
 };
 
 const getInitialValues = ({
   fromExample,
   fromCopy,
   allowAdvancedScheduling,
   variables,
+  searchParams,
 }: GetInitialValuesParams) => {
   let initialValues = defaultInitialValues;
 
+  // Will assume the query param has a valid ProvisionerType, as this query param is only used
+  // in testing.
+  defaultInitialValues.provisioner_type =
+    (searchParams.get("provisioner_type") as ProvisionerType) || "terraform";
```

Comment:
I'd kind of rather leave this as a `string`, because we're not doing any validation on it, so it could be anything

Reply:
You mean make `provisioner_type` a string?

Reply:
I think the `.get` here returns `string | undefined`, and the `||` narrows it to `string`, but I think the cast here narrow it more than it should, basically. we don't actually know that the returned value is a valid `ProvisionerType` here because we didn't check!

Reply:
Yes, that is correct. The type of the `defaultInitialValues.provisioner_type` is `ProvisionerType` which I would like to keep. Essentially I think adding validation is probably the best approach, but since this is a hidden query param only used in testing, I just skipped it for now. The BE will return a validation error if someone discovers this and tries to do something unsupported.